command/agent/agent.go

```diff
@@ -580,7 +580,7 @@ func (a *Agent) setupServer() error {
 		httpServ := &structs.Service{
 			Name:      a.config.Consul.ServerServiceName,
 			PortLabel: a.config.AdvertiseAddrs.HTTP,
-			Tags:      []string{consul.ServiceTagHTTP},
+			Tags:      append(a.config.Consul.Tags, consul.ServiceTagHTTP),
```

Comment:
`append` may actually modify the first array if it has capacity, I'd suggest doing the following here and subsequent Tag calls:

```suggestion
			Tags:      append([]string{consul.ServiceTagHTTP}, a.config.Consul.Tags),
```